src/parser/parser_support.c

```diff
@@ -11792,19 +11947,19 @@ pt_convert_dml (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continu
   switch (node->node_type)
     {
     case PT_INSERT:
-      pt_convert_dblink_insert_query (parser, node, node->sql_user_text, snl);
+      pt_convert_dblink_insert_query (parser, node, snl);
```

Comment:
`snl` does not require a return value from the pt_convert_dml function.
pt_convert_dblink_insert_query, pt_convert_dblink_delete_query, and pt_convert_dblink_update_query also do not use `snl`.
It seems only needed inside the pt_convert_dblink_dml_query function.

The pt_convert_dblink_merge_query function uses `snl`, but I don't know if I should use it only here.

Can we change it to use `snl` only in the pt_convert_dblink_dml_query function?

Reply:
Ok, I will execlude snl arguments from insert/delete/update except merge.

Reply:
I was mistaken. There seems to be no problem with `snl`.

Reply:
Get `snl` from pt_rewrite_for_dblink function.
But `snl` is empty until we pass it to the pt_convert_dblink_dml_query function.
```
pt_rewrite_for_dblink (parser, statement);
  -> SERVER_NAME_LIST snl;
  -> memset (&snl, 0x00, sizeof (SERVER_NAME_LIST)); /********** `snl` is empty. /**********/
  -> parser_walk_tree (parser, stmt, NULL, NULL, pt_convert_dml, &snl);
       -> pt_convert_dml
            -> pt_convert_dblink_insert_query (parser, node, snl); /********** Until here, `snl` is not used. /**********/
                 -> pt_convert_dblink_dml_query (parser, node, (remote_ins == 0), remote_ins, snl);
            -> pt_convert_dblink_delete_query (parser, node, snl); /********** Until here, `snl` is not used. /**********/
                 -> pt_convert_dblink_dml_query (parser, node, local_del, remote_del, snl);
            -> pt_convert_dblink_update_query (parser, node, snl); /********** Until here, `snl` is not used. /**********/
                 -> pt_convert_dblink_dml_query (parser, node, local_upd, remote_upd, snl);
            -> pt_convert_dblink_merge_query (parser, node, snl);
                 -> if (remote_target) 
                      {
                        parser_walk_tree (parser, source, pt_get_server_name_list, snl, NULL, NULL);
                        if (snl->local_cnt > 0)
                          {
                            /* not allowed to merge multi-server */
                            PT_ERROR (parser, source, "dblink: multi-remote DML is not allowed");
                          }
                        else
                          {
                            /* only remote merge */
                            assert (snl->server_cnt > 0);
                          }
                      }
                 -> pt_convert_dblink_dml_query (parser, node, (int) (remote_target == false), (int) (remote_target == true), snl);
```

Only the pt_convert_dblink_merge_query function is using `snl`.

`snl` does not require a return value from the pt_rewrite_for_dblink function.

No matter how much I think about it, I think `snl` continues to be passed unnecessarily.